packages/http/libraries/core/src/http/pipe/model/Pipe.ts

```diff
@@ -0,0 +1,7 @@
+import { HttpResponse } from '../../responses/HttpResponse';
+import { PipeMetadata } from './PipeMetadata';
+
+export interface Pipe<TInput = unknown, TOutput = unknown> {
+  execute(input: TInput, metadata: PipeMetadata): Promise<TOutput> | TOutput;
+  getHttpResponse?(): HttpResponse;
```

Comment:
I see we're applying the same pattern to the guard.

What if we want to set different status codes / messages depending on the received input?

We can maybe merge now and think how to solve this use case for guards / pipes in another PR